src/prefect/artifacts.py

```diff
@@ -0,0 +1,148 @@
+    )
+
+
+def update_link(task_run_artifact_id: str, link: str) -> None:
+    """
+    Update an existing link artifact. This function will replace the current link
+    artifact with the new link provided.
+
+    Note: The functionality here is experimental, and may change between
+    versions without notice. Use at your own risk.
+
+    Args:
+        - task_run_artifact_id (str): the ID of an existing task run artifact
+        - link (str): the new link to update the artifact with
+    """
+    if not running_with_backend():
+        return
+
+    if task_run_artifact_id is None:
+        raise ValueError("The ID of an existing task run artifact must be provided.")
+
+    client = Client()
+    client.update_task_run_artifact(
+        task_run_artifact_id=task_run_artifact_id, data={"link": link}
+    )
+
+
+def delete_link(task_run_artifact_id: str) -> None:
+    """
+    Delete an existing link artifact
+
+    Note: The functionality here is experimental, and may change between
+    versions without notice. Use at your own risk.
+
+    Args:
+        - task_run_artifact_id (str): the ID of an existing task run artifact
+    """
+    if not running_with_backend():
+        return
+
+    if task_run_artifact_id is None:
+        raise ValueError("The ID of an existing task run artifact must be provided.")
+
+    client = Client()
+    client.delete_task_run_artifact(task_run_artifact_id=task_run_artifact_id)
+
+
+def create_markdown(markdown: str) -> Optional[str]:
+    """
+    Create a markdown artifact
+
+    Note: The functionality here is experimental, and may change between
+    versions without notice. Use at your own risk.
+
+    Args:
+        - markdown (str): the markdown to post
+
+    Returns:
+        - str: the task run artifact ID
+    """
+    if not running_with_backend():
+        return None
+
+    client = Client()
+    return client.create_task_run_artifact(
+        task_run_id=context.get("task_run_id"),
+        kind="markdown",
+        data={"markdown": markdown},
+    )
+
+
+def update_markdown(task_run_artifact_id: str, markdown: str) -> None:
+    """
+    Update an existing markdown artifact. This function will replace the current markdown
+    artifact with the new markdown provided.
+
+    Note: The functionality here is experimental, and may change between
+    versions without notice. Use at your own risk.
+
+    Args:
+        - task_run_artifact_id (str): the ID of an existing task run artifact
+        - markdown (str): the new markdown to update the artifact with
+    """
+    if not running_with_backend():
+        return
+
+    if task_run_artifact_id is None:
+        raise ValueError("The ID of an existing task run artifact must be provided.")
+
+    client = Client()
+    client.update_task_run_artifact(
+        task_run_artifact_id=task_run_artifact_id, data={"markdown": markdown}
+    )
+
+
+def delete_markdown(task_run_artifact_id: str) -> None:
```

Comment:
Since all the delete functions are the same (just take a `task_run_artifact_id`), do we really need/want multiple versions or would a single `delete`/`delete_artifact` function be sufficient? If we want a `delete_*` method for each type, could at least share the common implementation in a `_` prefixed helper and call that in e.g. `delete_markdown`.

Reply:
I'm cool with a single delete artifact call 👍